radix-engine-macros/src/decimal.rs

```diff
@@ -0,0 +1,83 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Expr};
+
+extern crate radix_engine_common;
+
+fn get_decimal_from_expr(expr: &Expr) -> radix_engine_common::math::Decimal {
+    match expr {
+        Expr::Lit(lit) => match &lit.lit {
+            syn::Lit::Str(lit_str) => {
+                radix_engine_common::math::Decimal::try_from(lit_str.value()).unwrap()
+            }
+            syn::Lit::Int(lit_int) => {
+                radix_engine_common::math::Decimal::try_from(lit_int.base10_digits()).unwrap()
+            }
+            syn::Lit::Bool(lit_bool) => radix_engine_common::math::Decimal::from(lit_bool.value),
+            _ => panic!("Unsupported literal type!"),
```

Comment:
In all of these panics, we shouldn't panic, we should return an error with the span of the literal, in order to get the right behaviour in the IDE.

EG:
```rust
other_literal => return Err(Error::new(other_literal.span, "This macro only supports string and integer literals."));
```

Reply:
Just working on this after chat with Yulong :)